### DISCUSSION TITLE
Vanilla extract with nested providers

### DISCUSSION THREAD

**Discussion by tonai** [mantinedev/mantine#5688] [Q&A]
Hello,
is there a way to use vanilla extract with nested providers ?
I mean, when using nested providers, is there a way to get the correct variables of the nested theme ?
Bacuase in that case I can't just `import { vars } from './theme';` because I don't know what is the current nested theme.

**Comment by rtivital** [comment]
`vars` object contains reference to CSS variables. Variables updates are managed by the browser, you do not need to do anything on your side.

**Comment by tonai** [comment]
Thanks for your quick answer.
I was wondering why we should pass a `theme` object to the `themeToVars` function.
If the variable names are the same between the themes, and that's is the case for most of them, then there is no problems.
If you have a main theme and a nested subtheme, but if your subtheme has additionnal variables declared, then you can't access theme with `vars` (if vars was build using the main theme).
I mean we can't really access the "current" theme like with `useMantinetheme` inside `*.css.ts` files.
This is the same for `smallerThan` and `largerThan` helper functions in the `vars` object, they will still reference the value of the breakpoints of the main theme, and not the value of the "current" theme.
Is that correct ?

**Reply by rtivital** [reply]
To merge theme overrides and pass them to your vanilla extract styles, use https://mantine.dev/theming/theme-object/#merge-multiple-theme-overrides

Vanilla extract cannot handle dynamic values that are declared in React context.

**Reply by tonai** [reply]
Ok thank you.
In fact I was looking for a way to be able to do like responsive styles like in v6 with something like `theme.fn.smallerThan('lg')`.
I looked at the documentation about vanilla-extract and saw `vars.largerThan('sm')` so I was wondering if it was the same thing, but in fact not exactly.
My initial problem is that I have some kind of shared library that use Mantine, but I still want to allow the final consuming app of my library to customize the theme.
In that case I can't use something like `vars.largerThan('sm')` in the library that will use the final app theme value.
If I want to build some kind of reusable responsive component, I probably need to use some Js solution with `useMantinetheme` to get the current theme breakpoint values or is there another way ?
Thanks again.

**Reply by rtivital** [reply]
Dynamic breakpoint values that depend on the context are not really supported. There most likely will be other issues related to that. For example, with `hiddenFrom` and `visibleFrom` props. If you want to build a components' library that depends on Mantine breakpoints, the solution is either to use the same breakpoint values or switch to container queries.

**Reply by tonai** [reply]
Yes thank you I landed to the same conclusion as you.
By the way thank you for the great work.
And for information I did write an article about React UI libraries (here: https://blog.smile.eu/en/digital/best-react-ui-library) and I think that Mantine is really great :)
